src/Ormolu/Utils/Fixity.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Ormolu.Utils.Fixity
+  ( getFixityOverridesForSourceFile,
+    parseFixityDeclarationStr,
+  )
+where
+
+import Control.Exception (throwIO)
+import Control.Monad.IO.Class
+import Data.Bifunctor (first)
+import Data.IORef
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+import Ormolu.Exception
+import Ormolu.Fixity
+import Ormolu.Fixity.Parser
+import Ormolu.Utils.Cabal
+import System.Directory
+import System.FilePath
+import Text.Megaparsec (errorBundlePretty)
+
+-- | Attempt to locate and parse a @.ormolu@ file. If it does not exist,
+-- empty fixity map is returned. This function maintains a cache of fixity
+-- overrides where cabal file paths act as keys.
+getFixityOverridesForSourceFile ::
+  MonadIO m =>
+  -- | 'CabalInfo' already obtained for this source file
+  CabalInfo ->
+  m FixityMap
+getFixityOverridesForSourceFile CabalInfo {..} = liftIO $ do
+  cacheRef <- newIORef Map.empty
```

Comment:
This `cacheRef` is only used locally in this function (in particular, the subsequent `readIORef cacheRef` will always result in `Map.empty`). This should probably passed in here as an argument in some way?